server/tracedb/datastoreresource/repository_test.go

```diff
@@ -0,0 +1,499 @@
+package datastoreresource_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/gorilla/mux"
+	"github.com/kubeshop/tracetest/server/pkg/id"
+	"github.com/kubeshop/tracetest/server/resourcemanager"
+	rmtests "github.com/kubeshop/tracetest/server/resourcemanager/testutil"
+	"github.com/kubeshop/tracetest/server/testmock"
+	datastore "github.com/kubeshop/tracetest/server/tracedb/datastoreresource"
+	"github.com/stretchr/testify/require"
+)
+
+func compareJSONDataStores(t require.TestingT, operation rmtests.Operation, firstValue, secondValue string) {
+	if operation == rmtests.OperationUpdateSuccess {
+		require.JSONEq(t, firstValue, secondValue)
+		return
+	}
+
+	expected := rmtests.RemoveFieldFromJSONResource("createdAt", firstValue)
+	obtained := rmtests.RemoveFieldFromJSONResource("createdAt", secondValue)
+
+	require.JSONEq(t, expected, obtained)
+}
+
+func registerManagerFn(router *mux.Router) resourcemanager.Manager {
+	db := testmock.CreateMigratedDatabase()
+	dataStoreRepository := datastore.NewRepository(db)
+
+	manager := resourcemanager.New[datastore.DataStore](
+		datastore.ResourceName,
+		datastore.ResourceNamePlural,
+		dataStoreRepository,
+		resourcemanager.WithOperations(datastore.Operations...),
+		resourcemanager.WithIDGen(id.GenerateID),
+	)
+	manager.RegisterRoutes(router)
+
+	return manager
+}
+
+func cleanup(t *testing.T, manager resourcemanager.Manager) {
```

Comment:
Without a cleanup structure, this test file surpasses more than 100 idle connections. Later we need to think about how we can improve the RM structure to avoid these open connections.